src/proto.rs

```diff
@@ -327,18 +327,17 @@ pub(crate) struct Decoded<'a, T> {
     pub(crate) next: &'a [u8],
 }
 
-pub(crate) fn hash_mpint_bytes(hashed_secret: &[u8], exchange: &mut digest::Context) {
+pub(crate) fn hash_mpint_bytes(int: &[u8], exchange: &mut digest::Context) {
     // Remove leading zeros, and prepend a zero byte if the first byte has its
     // most significant bit set.
 
-    let leading_zeros = hashed_secret.iter().take_while(|&&b| b == 0).count();
-    if let Some(hashed_secret) = hashed_secret.get(leading_zeros..) {
-        let prepend = matches!(hashed_secret.first(), Some(&b) if b & 0x80 != 0);
-        let len = hashed_secret.len() + if prepend { 1 } else { 0 };
-        exchange.update(&(len as u32).to_be_bytes());
-        if prepend {
-            exchange.update(&[0]);
-        }
-        exchange.update(hashed_secret);
+    let leading_zeros = int.iter().take_while(|&&b| b == 0).count();
+    let int = &int[leading_zeros..];
```

Comment:
Can you add a comment about the slice being safe? I was trying to sort of statically avoid panicking.